### DISCUSSION TITLE
Better environment configuration

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#153] [Ideas]
.env files support limited formats

For example, a list or dict configuration is required.

Consider using dynaconf: [https://github.com/dynaconf/dynaconf](https://github.com/dynaconf/dynaconf)

**Comment by wu-clan** [comment]
env is simply an environment variable, just like you would configure it in the system environment variables, no more format

For other formats, it is now configured in Settings, which guarantees the data format through pydantic validation

The use of py as a configuration item is heavily influenced by django

---

Common profile options：toml > yaml > ini > json

Popular in python：ini -> yaml -> toml

---

Configuration via pydantic is currently popular, for dynaconf, check https://github.com/dynaconf/dynaconf/discussions/608 and consider switching after version 4.0.x

**Reply by downdawn** [reply]
OK.
